File: OpenLibraryBackend/src/main/java/com/open/library/POJO/Author.java
package com.open.library.POJO;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.DynamicInsert;
import org.hibernate.annotations.DynamicUpdate;

@Data
@AllArgsConstructor
@NoArgsConstructor
@DynamicInsert
@DynamicUpdate
@Entity
@Table(name = "authors")
public class Author {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "author_id")
    private Long id;

    private String fullName;

    @Column(columnDefinition = "MEDIUMBLOB")
    private String image;

    @Column(columnDefinition = "TEXT")
    private String story;

    private boolean is_activated;

    private boolean is_deleted;
}

File: OpenLibraryFrontend/src/app/shared/models/Author.ts
export class Author {
    id! :number;
    fullName? :string;
    story? :string;
    is_activated?: boolean;
    is_deleted?: boolean;

    constructor(id: number, fullName: string, story: string, is_activated: boolean, is_deleted: boolean) {
        this.id = id;
        this.fullName = fullName;
        this.story = story;
        this.is_activated = is_activated;
        this.is_deleted = is_deleted;
    }
}